adoc/chapters/programming_interface.adoc

```diff
@@ -6114,14 +6114,14 @@ if set final data is [code]#nullptr)#.
 [[subsec:mutex]]
 === Synchronization primitives
 
-When the user wants to use the [code]#buffer# simultaneously in the
-<<sycl-runtime>> and their own code (e.g. a multi-threaded mechanism) and wants
+When the user wants to use a [code]#buffer# simultaneously in the
+<<sycl-runtime>> (e.g. in a command running on a device) and their own code (e.g. a multi-threaded mechanism) and wants
```

Comment:
I think this is definitely better, but still not very clear.  I may be splitting hairs here, but developers can't use buffers in commands (they have to use accessors).  I'm also not sure that the multi-threaded example adds anything here, because the mutex is used to protect against race conditions between the runtime and user code rather than different threads in user code.

I'm not 100% if my suggestion below is accurate, but I think it would be clearer to write something like:
```suggestion
To prevent race conditions between accesses to the host allocation owned by a [code]#buffer# in the
<<sycl-runtime>> (e.g., by accessors) and in host code, it is necesssary
```

Reply:
Agree with your change! 
Do you think  we maybe use `hostData` in place of `host allocation`? 
For me `host allocation` mean `sycl::malloc_host`.  

But Thank to the `owned by a buffer`, the current sentence  is already unambiguous so not a big deal

Reply:
Note that internal issue 426 also points out that `use_mutex` is underspecified.  If you resolve that issue with this PR, please close it.

By my reading of the spec, it would be legal for the implementation to lock the mutex when the buffer is constructed and unlock it only when the buffer is destroyed.  Of course, this wouldn't be very useful, though.

I think we either need to define some guarantees on when the mutex is unlocked by the RT, or we need to deprecate / remove this property.

Reply:
> By my reading of the spec, it would be legal for the implementation to lock the mutex when the buffer is constructed and unlock it only when the buffer is destroyed. Of course, this wouldn't be very useful, though.

I agree with your reading. I don't think we need to define additional guarantees.

I do agree that we may want to consider deprecating/removing the property, though, if nobody understands what the feature is for.